label/label.go

```diff
@@ -191,11 +191,11 @@ func ImportPathToBazelRepoName(importpath string) string {
 	importpath = strings.ToLower(importpath)
 	components := strings.Split(importpath, "/")
 	labels := strings.Split(components[0], ".")
-	var reversed []string
+	var reversed = make([]string, 0, len(labels))
 	for i := range labels {
 		l := labels[len(labels)-i-1]
 		reversed = append(reversed, l)
 	}
 	repo := strings.Join(append(reversed, components[1:]...), "_")
-	return strings.NewReplacer("-", "_", ".", "_").Replace(repo)
+	return regexp.MustCompile(`\W`).ReplaceAllString(repo, "_")
```

Comment:
What name would you suggest for the lifted `regexp.Regexp`?

This is not backwards compatible, currently `github.com/user/go-.repo -> com_github_user_go__repo`. Is that OK?

I will add a unit test.

Reply:
For the lifted regexp, maybe `nonWordRe`? It'll be close, so it doesn't have to be too verbose.

About compatibility, I think it's okay. A change here should only affect newly added rules or labels without corresponding `go_repository`. We try to match on existing `go_repository` rules' `importpath` attributes before we generate a new name.

Reply:
It was also required to change [`strings.Join`](https://github.com/bazelbuild/bazel-gazelle/pull/688/files#diff-b17e01e1108a9f33eee647b3cd0824a4R201), to a character that would be replaced, to prevent `ht_sr_git_~sircmpwn_go-example -> ht_sr_git__sircmpwn_go_example`. Feel free to suggest a better character than `.`.